app/controllers/voters_controller.rb

```diff
@@ -64,44 +65,46 @@ def create
     end
 
     def update
-      if !admin_logged_in?
-        redirect_to "/"
+      # could allow for (timing based) Voter enumeration
+      @voter = Voter.find(params[:id])
+
+      unless can? :manage, GrantsVoter.new(voter: @voter)
```

Comment:
ok cool, I like where this is going.  Stretch Goal: Is there any way we can make a shim class that's like (pardon the Go code, that's what my head is in these days):

```
type User struct {
  Admin *Admin
  Voter *GrantsVoter
  Artist *Artist
}
```

And then the idea is that one or more of the members could be filled in, since right now more than one type of user could be logged in.  That would mean that the various can-checks would have to retry all the checks on each non-nil member, and return true if any of them satisfy the check.  

Having that structure would allow something along the lines of:

```
@voter = Voter.find(params[:id])
@user = UserFromVoter(@voter)  # Creates a new User object, only sets the Voter member
unless can? :manage, @user
```

We would have a bunch of helper functions to create User structs in various situations.

With that, there's a clear search-and-replace path to the eventual final form when we swap out the back end:

```
@user = User.find(params[:id])
unless can? :manage, @user
```

This would be my most idealist situation, but if that would cause headaches then we don't have to go that way.  What you've got here is good enough.